nva-datamodel-java/src/main/java/no/unit/nva/model/Publication.java

```diff
@@ -402,6 +409,30 @@ public void setCuratingInstitutions(Set<URI> curatingInstitutions) {
         this.curatingInstitutions = curatingInstitutions;
     }
 
+    public List<ImportDetail> getImportDetails() {
+        return nonNull(importDetails) ? importDetails : Collections.emptyList();
+    }
+
+    public void setImportDetails(List<ImportDetail> importDetails) {
+        if (importDetails == null || !new HashSet<>(importDetails).containsAll(getImportDetails())) {
+            throw new IllegalArgumentException(MUST_PRESERVE_EXISTING_IMPORT_DETAILS);
+        }
+
+        this.importDetails = new ArrayList<>(importDetails);
+    }
+
+    public void addImportDetail(ImportDetail importDetail) {
+        if (importDetail == null) {
+            return;
+        }
+
+        if (isNull(importDetails)) {
+            importDetails = new ArrayList<>();
+        }
+
+        importDetails.add(importDetail);
+    }
```

Comment:
You are using both isNull() and something == null, maybe use isNull() only :)

Reply:
Wops, will fix